tests/modules/text_generation/test_peft_tgis_remote.py

```diff
@@ -0,0 +1,103 @@
+        # Also, save the name of the dir + prompt ID, which is the path TGIS expects for the prefix ID
+        with tempfile.TemporaryDirectory() as model_dir:
+            causal_lm_dummy_model.save(model_dir)
+            mock_tgis_model = PeftPromptTuningTGIS.load(model_dir, StubTGISBackend())
+            model_prompt_dir = os.path.split(model_dir)[-1]
+
+        # Run an inference request, which is wrapped around our mocked Generate call
+        result = mock_tgis_model.run(SAMPLE_TEXT, preserve_input_text=True)
+        assert isinstance(result, GeneratedTextResult)
+        assert result.generated_text == "moose"
+        assert result.generated_tokens == 1
+        assert result.finish_reason == 5
+
+        stub_generation_request = mock_gen.call_args_list[0].args[0]
+
+        # Validate that our verbalizer carried over correctly & was applied at inference time
+        assert mock_tgis_model.verbalizer == causal_lm_dummy_model.verbalizer
+        assert stub_generation_request.requests[
+            0
+        ].text == "hello distributed {}".format(SAMPLE_TEXT)
+
+        # Ensure that our prefix ID matches the expected path based on our tmpdir and config
+        assert model_prompt_dir == stub_generation_request.prefix_id
+
+
+def test_load_and_run_stream_out(causal_lm_dummy_model):
+    """Ensure we can export an in memory model, load it, and (mock) run output streaming
+    with the right text & prefix ID."""
+    # Patch our stub backend into caikit so that we don't actually try to start TGIS
+    causal_lm_dummy_model.verbalizer = "hello distributed {{input}}"
+
+    with mock.patch.object(StubTGISClient, "GenerateStream") as mock_gen:
+        mock_gen.side_effect = StubTGISClient.stream_generate
+
+        # Save the local model & reload it a TGIS backend distributed module
+        # Also, save the name of the dir + prompt ID, which is the path TGIS expects for the prefix ID
+        with tempfile.TemporaryDirectory() as model_dir:
+            causal_lm_dummy_model.save(model_dir)
+            mock_tgis_model = PeftPromptTuningTGIS.load(model_dir, StubTGISBackend())
+            model_prompt_dir = os.path.split(model_dir)[-1]
+
+        # Run an inference request, which is wrapped around our mocked GenerateStream call
+        stream_result = mock_tgis_model.run_stream_out(
+            SAMPLE_TEXT, preserve_input_text=True
+        )
+        assert isinstance(stream_result, Iterable)
+        # Convert to list to more easily check outputs
+        result_list = list(stream_result)
+        assert len(result_list) == 3
+        first_result = result_list[0]
+        assert first_result.generated_text == "moose"
+        assert first_result.tokens[0].text == "moose"
+        assert first_result.tokens[0].logprob == 0.2
+        assert first_result.details.finish_reason == 5
+        assert first_result.details.generated_tokens == 1
+        assert first_result.details.seed == 10
```

Comment:
Can you move these checks to the fixtures too so that we can just `.validate_unvary_generate_response` / `.validate_generate_stream_response` here + in the text generation tests?